nebula/ui/3rdparty/lottie/LottieAnimation.qml

```diff
@@ -0,0 +1,505 @@
+
+            destroyAnimation();
+            playIfShould();
+        }
+
+        function destroyAnimation() {
+            if (animationItem) {
+                animationItem.destroy();
+                animationItem = null;
+            }
+            lottieItem.clear();
+        }
+
+        function updateAnimationSize() {
+            if (animationItem) {
+                lottieItem.clear();
+                animationItem.resize();
+            }
+        }
+
+        Component.onCompleted: {
+            componentComplete = true;
+        }
+    }
+
+    // Should we move these handlers into a Connections {} within the private?
+    onRunningChanged: {
+        if (running) {
+            start();
+        } else {
+            pause();
+        }
+    }
+
+    onWidthChanged: Qt.callLater(d.updateAnimationSize)
+    onHeightChanged: Qt.callLater(d.updateAnimationSize)
+
+    // TODO Would be lovely if we could change those at runtime without recreating the animation
+    onLoopsChanged: Qt.callLater(d.destroyAndRecreate)
+    onClearBeforeRenderingChanged: Qt.callLater(d.destroyAndRecreate)
+    onFillModeChanged: Qt.callLater(d.destroyAndRecreate)
+
+    onSpeedChanged: {
+        if (d.animationItem) {
+            d.animationItem.setSpeed(speed);
+        }
+    }
+    onReverseChanged: {
+        if (d.animationItem) {
+            d.animationItem.setDirection(reverse ? -1 : 1);
+        }
+    }
+
+    onSourceChanged: {
+        // is already JS object, use verbatim
+        if (typeof source === "object") { // TODO what about QUrl, I think it is treated as {} here
+            console.log(d.log, "Using source verbatim as it is an object");
+            d.animationData = source;
+            return;
+        }
+
+        var sourceString = source.toString();
+
+        if (sourceString.indexOf("{") === 0) { // startsWith("{"), assume JSON
+            console.log(d.log, "Using source as JSON");
+            d.setAnimationDataJson(sourceString);
+            return;
+        }
+
+        d.animationData = null;
+        if (!source) {
+            return;
+        }
+
+        var url = source.toString(); // toString in case is QUrl
+        if (url.indexOf("/") === 0) { // assume local file
+            url = "file://" + url;
+        } else if (url.indexOf(":/") === -1) { // assume relative url
+            // FIXME figure out how to do relative URLs with Ajax
+            // Qt.resolvedUrl is relative to *this* file, not the one where the item is actually used from
+        }
+
+        // NOTE QML LoggingCategory {} has its internal QLoggingCategory created in
+        // componentCompleted(). There seems to be a situation where this console.log
+        // is executed before the LoggingCategory {} object above has completed.
+        //console.log(d.log, "Fetching source from", url);
+
+        var xhr = new XMLHttpRequest()
+        // FIXME allow asynchronous
+        xhr.open("GET", url, false /*synchronous*/);
```

Comment:
It's not good to load resources synchronously - this is ok because they are local resources, but in theory, they can be remote too.